Libraries/LibC/netdb.cpp

```diff
@@ -202,10 +212,57 @@ hostent* gethostbyaddr(const void* addr, socklen_t addr_size, int type)
     return &__gethostbyaddr_buffer;
 }
 
+//References for *servent implementations can be found here: https://pubs.opengroup.org/onlinepubs/007908799/xns/getservent.html
+//FIXME: description of getservbyname states that it gets the next entry of the database, opening a connection if necessary.
+//This to me says that the line to begin reading on should be saved even if connection is closed.
+//Also need to factor in the persistent connection provided byt setservent
 struct servent* getservent()
 {
-    dbg() << "FIXME: getservbyname()";
-    ASSERT_NOT_REACHED();
+    //If the services file is not open, attempt to open it and return null if it fails.
+    if(!services_file) {
```

Comment:
Please use `clang-format` as you go instead of doing it in a separate commit at the end.